drivers/ph_oem/ph_oem.c

```diff
@@ -0,0 +1,446 @@
+        i2c_release(I2C);
+        return PH_OEM_WRITE_ERR;
+    }
+
+    /* Calibration is critical, so check if written value is in fact correct */
+    if (i2c_read_regs(I2C, ADDR, PH_OEM_REG_CALIBRATION_BASE, &reg_value, 4, 0) < 0) {
+        DEBUG("\n[ph_oem debug] Calibrating device failed \n");
+        i2c_release(I2C);
+        return PH_OEM_READ_ERR;
+    }
+
+    uint16_t confirm_value = (int16_t)(reg_value[2] << 8)
+                             | (int16_t)(reg_value[3]);
+
+    if (confirm_value != calibration_value) {
+        DEBUG("\n[ph_oem debug] Calibrating device to pH raw %d failed \n",
+              calibration_value);
+        i2c_release(I2C);
+        return PH_OEM_WRITE_ERR;
+    }
+
+    i2c_release(I2C);
+
+    return PH_OEM_OK;
+}
+
+int ph_oem_set_calibration(const ph_oem_t *dev, uint16_t calibration_value,
+                           ph_oem_calibration_option_t option)
+{
+    if (ph_oem_set_calibration_value(dev, calibration_value) != PH_OEM_OK) {
+        return PH_OEM_WRITE_ERR;
+    }
+
+    uint8_t reg_value;
+
+    i2c_acquire(I2C);
+
+    if (i2c_write_reg(I2C, ADDR, PH_OEM_REG_CALIBRATION_REQUEST,
+                      option, 0) < 0) {
+        DEBUG("\n[ph_oem debug] Sending calibration request failed\n");
+        return PH_OEM_WRITE_ERR;
+    }
+
+    do {
+        if (i2c_read_reg(I2C, ADDR, PH_OEM_REG_CALIBRATION_REQUEST, &reg_value,
+                         0) < 0) {
+            DEBUG("\n[ph_oem debug] Reading calibration request status failed\n");
+            i2c_release(I2C);
+            return PH_OEM_READ_ERR;
+        }
+    } while (reg_value != 0x00);
+
+    i2c_release(I2C);
+
+    return PH_OEM_OK;
+}
+
+int ph_oem_read_calibration_state(const ph_oem_t *dev, int16_t *calibration_state)
+{
+    i2c_acquire(I2C);
+
+    if (i2c_read_reg(I2C, ADDR, PH_OEM_REG_CALIBRATION_CONFIRM,
+                     calibration_state, 0) < 0) {
+        DEBUG("\n[ph_oem debug] Failed at reading calibration confirm register\n");
+        i2c_release(I2C);
+        return PH_OEM_READ_ERR;
+    }
+    i2c_release(I2C);
+    return PH_OEM_OK;
+}
+
+int ph_oem_set_compensation(const ph_oem_t *dev,
+                            uint16_t temperature_compensation)
+{
+    uint8_t reg_value[4];
+
+    reg_value[0] = 0x00;
+    reg_value[1] = 0x00;
+    reg_value[2] = (uint8_t)(temperature_compensation >> 8);
+    reg_value[3] = (uint8_t)(temperature_compensation & 0x00FF);
+
+    i2c_acquire(I2C);
+
+    if (i2c_write_regs(I2C, ADDR, PH_OEM_REG_TEMP_COMPENSATION_BASE,
+                       &reg_value, 4, 0) < 0) {
+        DEBUG("\n[ph_oem debug] Setting temperature compensation of device to "
+              "%d failed\n", temperature_compensation);
+        i2c_release(I2C);
+        return PH_OEM_WRITE_ERR;
+    }
+    i2c_release(I2C);
+
+    return PH_OEM_OK;
+}
+
+int ph_oem_read_compensation(const ph_oem_t *dev,
+                             int16_t *temperature_compensation)
```

Comment:
Why `int16_t`? The temperature compensation value is always positive.

Reply:
changed int16_t to uint16_t